### ISSUE TITLE
[Regression][5.71] Unable to connect Dualshock3

### ISSUE THREAD

**Issue by mmbossoni** [bluez/bluez#688]
Before bluez 5.71:
1 - Connect dualshock 3 through usb cable
2 - bluez would ask trust + authorize
3 - disconnect cable
4 - press PS button and you are device is connected

With bluez 5.71:
On Step 4, when pressing the PS button, bluez asks for a PIN even though device doesn't require one.

I've made sure PS3 plugin is installed.

Let me know which kind of debug info is needed

output of bluetoothctl
```
Waiting to connect to bluetoothd...
[bluetooth]# Agent registered
[bluetooth]# default-agent 
[bluetooth]# Default agent request successful
[bluetooth]# devices 
[bluetooth]# [NEW] Device 38:C0:96:56:4D:A0 Sony PLAYSTATION(R)3 Controller
[bluetooth]# Authorize service
[bluetooth]#   1;39m[agent] Authorize service 00001124-0000-1000-8000-00805f9b34fb (yes/no): yes
[bluetooth]# [CHG] Device 38:C0:96:56:4D:A0 UUIDs: 00001124-0000-1000-8000-00805f9b34fb
[bluetooth]# [CHG] Device 38:C0:96:56:4D:A0 WakeAllowed: yes
[bluetooth]# [CHG] Device 38:C0:96:56:4D:A0 Class: 0x00000508 (1288)
[bluetooth]# [CHG] Device 38:C0:96:56:4D:A0 Icon: input-gaming
[CHG] Device 38:C0:96:56:4D:A0 Connected: yestroller
[Sony PLAYSTATION(R)3 Controller]# Request PIN code
[Sony PLAYSTATION(R)3 Controller]#   1;39m[agent] Enter PIN code: 0000
[Sony PLAYSTATION(R)3 Controller]# [CHG] Device 38:C0:96:56:4D:A0 ServicesResolved: no
[CHG] Device 38:C0:96:56:4D:A0 Connected: noluetooth
[CHG] Device 38:C0:96:56:4D:A0 Connected: yestroller
[Sony PLAYSTATION(R)3 Controller]# Request PIN code
[Sony PLAYSTATION(R)3 Controller]#   1;39m[agent] Enter PIN code: Request canceled
[agent] Enter PIN code:   0;94m[bluetooth]# [CHG] Device 38:C0:96:56:4D:A0 Connected: no
[bluetooth]# 

``` 

[bluetoothd.log](https://github.com/bluez/bluez/files/13695841/bluetoothd.log)

**Comment by dmanlfc** [comment]
confirmed, same issue - `[Sony PLAYSTATION(R)3 Controller]# hci0 04:98:F3:76:A8:DD auth failed with status 0x05 (Authentication Failed)
[Sony PLAYSTATION(R)3 Controller]# Failed to pair: org.bluez.Error.AuthenticationFailed
[Sony PLAYSTATION(R)3 Controller]# hci0 04:98:F3:76:A8:DD type BR/EDR disconnected with reason 4
[CHG] Device 04:98:F3:76:A8:DD Connected: noluetooth
[bluetooth]# hci0 04:98:F3:76:A8:DD type BR/EDR connected eir_len 5
[CHG] Device 04:98:F3:76:A8:DD Connected: yestroller`

rolling back to 5.70 for now.

**Comment by mmbossoni** [comment]
After further testing, this might be a dup of https://github.com/bluez/bluez/issues/673, since changing the ClassicBondedOnly config fixed it for me

**Comment by kohane27** [comment]
Downgrading to 5.64-2 works for me.

**Comment by legluondunet** [comment]
> Downgrading to 5.64-2 works for me.

You do not need to downgrade bluez packages, on Arch, created input.conf file in /etc/bluetooth folder, then added this lines:

```
[General]
ClassicBondedOnly=false
```

restart bluetooth:
sudo systemctl restart bluetooth

my Dualshock 3 works again without pin code.

**Comment by xadips** [comment]
Yup the `ClassicBondedOnly=false` helps on multiple controllers since the default change

**Comment by KaroLaunonen** [comment]
Great! Finally solution for Dualshock 3. It's been broken for me since 5.68 but this fixes it on 5.72. Thanks!

**Comment by DavisNT** [comment]
Setting `ClassicBondedOnly=false` will also enable the CVE-2023-45866 security vulnerability.

For details see:
- https://ubuntu.com/security/CVE-2023-45866 
- https://git.kernel.org/pub/scm/bluetooth/bluez.git/commit/profiles/input?id=25a471a83e02e1effb15d5a488b3f0085eaeb675

**Comment by manuelafm** [comment]
FYI:
It should now be possible to connect the DualShock3 with Bluetooth with limited exposure to [CVE-2023-45866](https://ubuntu.com/security/CVE-2023-45866) and without changing `ClassicBondedOnly=false`: https://github.com/bluez/bluez/issues/1165
